pkg/apis/core/v1alpha1/types.go

```diff
@@ -258,6 +258,7 @@ type CDIConfig struct {
 type CDIConfigSpec struct {
 	UploadProxyURLOverride   *string `json:"uploadProxyURLOverride,omitempty"`
 	ScratchSpaceStorageClass *string `json:"scratchSpaceStorageClass,omitempty"`
+	PodResourceRequirements *corev1.ResourceRequirements `json:"podResourceRequirements,omitempty"`
```

Comment:
So this will be the override right? The pattern we have been using in the CDIConfig object is the following.

The status reflects what the import/upload/clone controllers will use when creating the pods. The spec contains optional overrides. So I am missing a 
'DefaultPodResourceRequirements' in the CDIConfigStatus struct. then we need to add something to the config-controller reconcile loop, that checks if the PodResourceRequirements exists, if it does, copy that to the status.DefaultPodResourceRequirements, if it doesn't then the status.DefaultPodResourceRequirements should be our 'reasonable default values'.

Reply:
>So this will be the override right?

Yes, that's the object we want to use for creating temporary pods.

>The status reflects what the import/upload/clone controllers will use when creating the pods. 

Before your review, we thought that importer/upload/clone controllers could use the values directly from CDIConfigSpec, not CDIConfigStatus.

So if I understand right, you mean that
1. There should be some functions that insert 'reasonable default values' in status.DefaultPodResourceRequirements when deploying cdi modules.
2. Also, in config-controller reconcile loop, when spec.PodResourceRequirements is filled, loop always copy that to the status.DefaultPodResourceRequirements.
3. At 'pkg/controller/util.go', we only use the values in status.DefaultPodResourceRequirements.

Is it right?

Reply:
> So if I understand right, you mean that
> 
>     1. There should be some functions that insert 'reasonable default values' in status.DefaultPodResourceRequirements when deploying cdi modules.
> 
This will be the config controller I think. We do something similar for the scratchspace and proxyurl already. If those don't exist in the spec, do some default behavior.


>     2. Also, in config-controller reconcile loop, when spec.PodResourceRequirements is filled, loop always copy that to the status.DefaultPodResourceRequirements.
> 

Yes, if the spec if filled copy those values into the status, and if there is a partial fill, then the rest will be the default values.

>     3. At 'pkg/controller/util.go', we only use the values in status.DefaultPodResourceRequirements.

Correct.
> 
> 
> Is it right?